test/test_little_dict.jl

```diff
@@ -0,0 +1,474 @@
+        @test OrderedCollections.isordered(LittleDict{Int, String})
+        @test !OrderedCollections.isordered(Dict{Int, String})
+    end
+
+    @testset "Test merging" begin
+        a = LittleDict("foo"  => 0.0, "bar" => 42.0)
+        b = LittleDict("フー" => 17, "バー" => 4711)
+        @test isa(merge(a, b), LittleDict{String,Float64})
+    end
+
+    @testset "Issue #9295" begin
+        d = LittleDict()
+        @test push!(d, 'a'=> 1) === d
+        @test d['a'] == 1
+        @test push!(d, 'b' => 2, 'c' => 3) === d
+        @test d['b'] == 2
+        @test d['c'] == 3
+        @test push!(d, 'd' => 4, 'e' => 5, 'f' => 6) === d
+        @test d['d'] == 4
+        @test d['e'] == 5
+        @test d['f'] == 6
+        @test length(d) == 6
+    end
+
+    @testset "Serialization" begin
+        s = IOBuffer()
+        od = LittleDict{Char,Int64}()
+        for c in 'a':'e'
+            od[c] = c-'a'+1
+        end
+        serialize(s, od)
+        seek(s, 0)
+        dd = deserialize(s)
+        @test isa(dd, OrderedCollections.LittleDict{Char,Int64})
+        @test dd == od
+        close(s)
+    end
+
+    @testset "Issue #148" begin
+        d148 = LittleDict(
+                    :gps => [],
+                    :direction => 1:8,
+                    :weather => 1:10
+            )
+
+        d148_2 = LittleDict(
+            :time => 1:10,
+            :features => LittleDict(
+                :gps => 1:5,
+                :direction => 1:8,
+                :weather => 1:10
+            )
+        )
+    end
+
+    @testset "Issue #400" begin
+        @test filter(p->first(p) > 1, LittleDict(1=>2, 3=>4)) isa LittleDict
+    end
+
+    @testset "Sorting" begin
+        d = LittleDict(i=>Char(123-i) for i in [4, 8, 1, 7, 9, 3, 10, 2, 6, 5])
+        
+        @test collect(keys(d)) != 1:10
+        sd = sort(d)
+        @test collect(keys(sd)) == 1:10
+        @test collect(values(sd)) == collect('z':-1:'q')
+        @test sort(sd) == sd
+        sdv = sort(d; byvalue=true)
+        @test collect(keys(sdv)) == 10:-1:1
+        @test collect(values(sdv)) == collect('q':'z')
+    end
+
+    @testset "Test that LittleDict merge with combiner returns type LittleDict" begin
+        @test merge(+, LittleDict(:a=>1, :b=>2), LittleDict(:b=>7, :c=>4)) == LittleDict(:a=>1, :b=>9, :c=>4)
+        @test merge(+, LittleDict(:a=>1, :b=>2), Dict(:b=>7, :c=>4)) isa LittleDict
+    end
+end # @testset LittleDict
+
+
+@testset "Frozen LittleDict" begin
+
+    @testset "types" begin
+        base_dict = LittleDict((10,20,30),("a", "b", "c"))
+        @test base_dict isa LittleDict{Int, String, <:Tuple, <:Tuple}
+        
+        nonfrozen = LittleDict(10=>"a", 20=>"b", 30=>"c")
+        @test nonfrozen isa LittleDict{Int, String, <:Vector, <:Vector}
+        
+        @test base_dict == nonfrozen
+
+        frozen = freeze(nonfrozen)
+        @test frozen == base_dict
+        @test frozen isa LittleDict{Int, String, <:Tuple, <:Tuple}
```

Comment:
Could also test `frozen === base_dict`.

Reply:
Should they `===` ?
They are different objects

Reply:
But they're fully immutable, so they should compare identically (and they do).

Reply:
great.